repository-postgresql/src/main/java/org/lfenergy/compas/scl/data/repository/postgresql/CompasSclDataPostgreSQLRepository.java

```diff
@@ -155,6 +155,29 @@ public String findByUUID(SclFileType type, UUID id, Version version) {
         }
     }
 
+    @Override
+    public boolean hasDuplicateSclName(SclFileType type, String name) {
+        var sql = "SELECT DISTINCT ON (id) m.* "
+                + "FROM scl_file m "
+                + "WHERE m.type=? "
+                + "ORDER BY m.id, m.major_version desc, m.minor_version desc, m.patch_version desc";
+
+        try (var connection = dataSource.getConnection();
+             var stmt = connection.prepareStatement(sql)) {
+            stmt.setString(1, type.name());
+
+            try (var resultSet = stmt.executeQuery()) {
+                while (resultSet.next()) {
+                    var usedName = resultSet.getString(NAME_FIELD);
+                    if (usedName.equals(name)) return true;
+                }
+            }
+            return false;
+        } catch (SQLException exp) {
+            throw new CompasSclDataServiceException(POSTGRES_SELECT_ERROR_CODE, "Error select meta info from database!", exp);
```

Comment:
The message is probably copied from a other method. :-)

Reply:
Already fixed it in latest commit, changed it to `Error selecting latest versions from database!`